### ISSUE TITLE
Add option of authorisation to smibhid webserver

### ISSUE THREAD

**Issue by sam57719** [somakeit/smib#253]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#the_general_http_authentication_framework

basicauth seems fairly doable

**Comment by sjefferson99** [comment]
Yeah should be able to do basic auth, can't encrypt easily, it's a load of load, but would stop people scanning the network for web sites and then messing with it. But as you can mess with it by plugging in a USB (not locked in a vending machine) I didn't raise this earlier as too urgent, but worth doing sometime.
